Code/Framework/AzCore/AzCore/Serialization/Json/MapSerializer.cpp

```diff
@@ -333,6 +333,33 @@ namespace AZ
             {
                 // Convert the array to a Object with the member being the key.
                 outputValue.SetObject();
+
+                if (sortResult)
+                {
+                    // RapidJSON has a feature enabled called RAPIDJSON_USE_MEMBERSMAP.
+                    // When defined, RapidJSON system stores a multimap in addition to the array of members
+                    // to speed up lookups so that find and erase operations can be O(logn) instead of linear.
+                    // However, because of this, sorting the array of members without updating the map (which has
+                    // no API to do that) will break the object's data structure.
+                    // so to keep these objects in order, but not call sort() on the actual constructed
+                    // json document, instead, sort first, before inserting.
+                    auto less = [](const rapidjson::Value& lhs,
+                                   const rapidjson::Value& rhs) -> bool
+                    {
+                        JsonSerializerCompareResult result = JsonSerialization::Compare(lhs, rhs);
+                        if (result == JsonSerializerCompareResult::Equal)
+                        {
+                            return JsonSerialization::Compare(lhs, rhs) == JsonSerializerCompareResult::Less;
+                        }
+                        else
+                        {
+                            return result == JsonSerializerCompareResult::Less;
+                        }
+                    };
+                    // Using std::sort because AZStd::sort isn't implemented in terms of move operations.
+                    std::sort(keyValues.Begin(), keyValues.End(), less);
```

Comment:
FYI: AZStd::sort is just an alias for https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/std/sort.h#L773-L774 now, so the comment is not accurate.